tajo-catalog/tajo-catalog-client/src/main/java/org/apache/tajo/catalog/AbstractCatalogClient.java

```diff
@@ -380,15 +409,11 @@ public final PartitionDescProto getPartition(final String databaseName, final St
           .setTableName(tableName)
           .setPartitionName(partitionName)
           .build();
-
       final GetPartitionDescResponse response = stub.getPartitionByPartitionName(null, request);
 
-      if (ReturnStateUtil.isThisError(response.getState(), ResultCode.UNDEFINED_PARTITION)) {
-        throw new UndefinedPartitionException(partitionName);
-      }
-
+      throwsIfThisError(response.getState(), UndefinedPartitionMethodException.class);
+      throwsIfThisError(response.getState(), UndefinedPartitionException.class);
```

Comment:
It seems that UndefinedDatabaseException and UndefinedTableExeption can also occur.